code/modules/projectiles/guns/energy/special.dm

```diff
@@ -214,13 +214,34 @@
 
 /obj/item/gun/energy/wormhole_projector
 	name = "bluespace wormhole projector"
-	desc = "A projector that emits high density quantum-coupled bluespace beams."
+	desc = "A projector that emits high density quantum-coupled bluespace beams. Requires a bluespace anomaly core to function."
 	ammo_type = list(/obj/item/ammo_casing/energy/wormhole, /obj/item/ammo_casing/energy/wormhole/orange)
 	item_state = null
 	icon_state = "wormhole_projector"
 	var/obj/effect/portal/p_blue
 	var/obj/effect/portal/p_orange
 	var/atmos_link = FALSE
+	var/firing_core
+
+/obj/item/gun/energy/wormhole_projector/attackby(obj/item/C, mob/user)
+	if(istype(C, /obj/item/assembly/signaler/anomaly/bluespace))
+		to_chat(user, "<span class='notice'>You insert [C] into the wormhole projector and the weapon gently hums to life.</span>")
+		firing_core = TRUE
+		playsound(src.loc, 'sound/machines/click.ogg', 50, TRUE)
+		qdel(C)
+	else
+		firing_core = FALSE

```

Comment:
this will re-disable the gun if you hit it with anything else afterwards

Reply:
Suggestions? Just leave out the else clause?

Reply:
put in `else ..()` in its place if you want to keep it as is

If you want to make the core removable, you can add another check for, say, a screwdriver, that would drop the core and set firing_core to false. You should be able to find something like that in a bunch of other objects or machinery as examples.

Reply:
I don't - there's no reason someone would ever need to remove the core.